hibernate-reactive-core/src/main/java/org/hibernate/reactive/pool/impl/Parameters.java

```diff
@@ -0,0 +1,76 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive.pool.impl;
+
+public class Parameters {
+
+	private boolean inString;
+	private boolean inQuoted;
+	private boolean inSqlComment;
+	private boolean inCComment;
+	private boolean escaped;
+	private int count = 0;
+	private StringBuilder result;
+	private int previous;
+
+	private Parameters(String sql) {
+		result = new StringBuilder( sql.length() + 10 );
```

Comment:
What's the 10 about?

Reply:
It's an estimate of how many parameters we're likely to have to expand. Obviously it's not a strict upper bound, especially in the case of `insert`s and `update`s. Perhaps I should increase it for inserts and updates.

(The purpose is to avoid having to grow the `StringBuilder`.)

Reply:
I did consider scanning the string for `?`s but it wasn't clear to me if doing this *every time*, this was faster or slower than *occasionally* growing the `StringBuilder`.

Reply:
I kinda guess that, but this is the kind of number that people in the future will wonder about if you don't add a comment to it

Reply:
Actually this is stupid. I actually know how many parameters I have, don't I?

Reply:
I've fixed it to use the exact number of parameters.